agent/consul/state/catalog_test.go

```diff
@@ -1950,31 +1939,40 @@ func TestStateStore_ServicesByNodeMeta(t *testing.T) {
 		if err != nil {
 			t.Fatalf("err: %s", err)
 		}
-		expected := structs.Services{
-			"redis": []string{"primary", "prod"},
-		}
-		sort.Strings(res["redis"])
-		require.Equal(t, expected, res)
+		require.Len(t, res, 1)
+		require.True(t, res[0].ServiceID == "service1" && res[0].Node == "node0")
```

Comment:
Something like this might be a bit more obvious, and give better error messages on failures. Does this work?

```suggestion
		require.Equal(t, res, []*structs.NodeServbice{ns1})
```

Reply:
Hrm, I'm not sure how to make this work. I like it, it's far cleaner. However, here we are receiving `[]*struct.ServiceNode`  but ns1, ns2 are defined as `structs.NodeService`.

(To be honest, the difference confuses me)

Reply:
Ah, oops, I missed they were different types. Ya I guess `ns1` and `ns2` won't work. Maybe something like this?

```go
 []*structs.NodeService{ns1.ToServiceNode("node0")}
```

The names do make it difficult to see the difference. One type is persisted and used for some endpoints, and the other type is only used for API/RPC endpoints.

Reply:
Sorry about the delay in getting back to you on this. I tried this solution but reflect.Equals is tripping on the structs' RaftIndex values changing. I'm not sure if you would want to pull in something like go-cmp/gotestyourself or create a custom comparison function.
I've committed the recommended changes, I like the approach. Just not sure what the best way to tackle it in keeping with the project.